File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.py
from collections import deque

def solution(cacheSize, cities):
    answer = 0
    cache = deque([])
    
    for i in cities:
        a = i.lower()
        if cacheSize == 0:
            answer += 5
            continue
        elif len(cache) < cacheSize and a not in cache:
            cache.append(a)
            answer += 5
            continue
        elif len(cache) < cacheSize and a in cache:
            answer += 1
            cache.remove(a)
            cache.append(a)
            continue
        else:
            if a in cache:
                answer += 1
                cache.remove(a)
                cache.append(a)
            else:
                answer += 5
                cache.popleft()
                cache.append(a)
            
    return answer

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.LinkedList;

class Solution {
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;
        
        if (cacheSize == 0) return 5 * cities.length;
        
        LinkedList<String> cacheList = new LinkedList<>();
        for (int i = 0; i < cities.length; i++) {
            if (!cacheList.contains(cities[i].toLowerCase())) {
                if (cacheList.size() >= cacheSize) cacheList.remove(cacheList.get(0));
                cacheList.add(cities[i].toLowerCase());
                answer += 5;
            } else {
                cacheList.remove(cities[i].toLowerCase());
                cacheList.add(cities[i].toLowerCase());
                answer++;
            }
        }
        
        return answer;
    }
}